erpnext/stock/doctype/stock_entry/test_stock_entry.py

```
@@ -2280,6 +2280,98 @@ def test_manufacture_entry_without_wo(self):
+		# Create items.  1 finished good, 4 raw mats.
+		fg_item = make_item("_Disassemble Mobile", properties={"is_stock_item": 1}).name
+		rm_item1 = make_item("_Disassemble Temper Glass", properties={"is_stock_item": 1, "valuation_rate": 1}).name
+		rm_item2 = make_item("_Disassemble Battery", properties={"is_stock_item": 1, "valuation_rate": 1}).name
+		rm_item3 = make_item("_Disassemble Speaker", properties={"is_stock_item": 1, "valuation_rate": 1}).name
+		rm_item4 = make_item("_Disassemble Alternative Battery", properties={"is_stock_item": 1, "valuation_rate": 1}).name
+
+		# Create BOM for finished good, using only first 2 raw mats.
+		bom_no = make_bom(item=fg_item, raw_materials=[rm_item1, rm_item2, rm_item3]).name
+
+		# Create a Work Order for finished good.  We want to make 20
+		work_order = frappe.new_doc("Work Order")
```

Ruff (0.15.0)

[error] `frappe` may be undefined, or defined from star imports

(F405)